addon/build-waiter.ts

```diff
@@ -0,0 +1,37 @@
+import { DEBUG } from '@glimmer/env';
+import { TestWaiter } from 'ember-test-waiters';
+import NoopTestWaiter from './noop-test-waiter';
+
+/**
+ * Builds and returns a test waiter. The type of the
+ * returned waiter is dependent on whether the app or
+ * addon is in `DEBUG` mode or not.
+ *
+ * @param name {string} The name of the test waiter
+ * @returns {TestWaiter|NoopTestWaiter}
+ *
+ * @example
+ *
+ * import Component from '@ember/component';
+ * import { buildWaiter } from 'ember-test-waiters';
+ *
+ * if (DEBUG) {
+ *   let waiter = buildWaiter('friend-waiter');
+ * }
+ *
+ * export default class Friendz extends Component {
+ *   didInsertElement() {
+ *     waiter.beginAsync(this);
+ *
+ *     someAsyncWork().then(() => {
+ *       waiter.endAsync(this);
+ *     });
+ *   }
+ * }
+ */
+export default function buildWaiter(name: string) {
```

Comment:
Can you add an explicit return type here? I know typescript can infer it, but when reading its nice to see...

Reply:
👍